src/exotic-sizes.md

```diff
@@ -106,9 +106,9 @@ consequence of types with no size. In particular, pointer offsets are no-ops,
 and allocators typically [require a non-zero size][alloc].
 
 Note that references to ZSTs (including empty slices), just like all other
-references, must be non-null and suitably aligned. Dereferencing a null or
-unaligned pointer to a ZST is [undefined behavior][ub], just like for any other
-type.
+references, must be non-null and suitably aligned. However, dereferencing a
+null pointer to a ZST is not [undefined behavior][ub], unlike pointers to
+other types.
```

Comment:
In fact dereferencing (as in, using the `*` operator on) a null pointer is never UB. `addr_of!(*ptr)` is always safe, even if `ptr` is null.

Only loads, stores, and field projections have any requirements.

Reply:
I've changed it to say "loading or storing through a null pointer to a ZST"